includes/class-theme-base.php

```diff
@@ -547,32 +564,51 @@ function post() {
 
 	/**
 	 * @return WPLib_Post_Base
-	 *
-	 * @future Make work for non-posts?
 	 */
 	function item() {
 
-		return $this->has_posts() 
-			? WPLib_Posts::make_new_item( $this->post() ) 
-			: new WPLib_Post_Default( null );
+		$item = null;
+		if( $this->has_posts() ) {
+			$_post = $this->post();
+			$item  = WPLib_Posts::make_new_item( $_post, "list_owner=" );
+			if( !$item ){
+				$item  = new WPLib_Post_Default( $_post );
+			}
+		} else {
+			$item  = new WPLib_Post_Default( null );
+		}
+
+		return $item;
 
 	}
 
 	/**
-	 * @return WPLib_Page
+	 * @return WPLib_Page|WPLib_Post_Base
 	 */
 	function page_item() {
 
-		return new WPLib_Page( $this->post() );
+		$_post = $this->post();
+
+		if( $_post && 'page' !== $_post->post_type ){
+			WPLib::trigger_error( sprintf( "Current post is of '%s' post type and should not be requested via \$theme->page_item()", $_post->post_type) );
+		}
+
+		return $this->item();
 
 	}
 
 	/**
-	 * @return WPLib_Post
+	 * @return WPLib_Post|WPLib_Post_Base
 	 */
 	function post_item() {
 
-		return new WPLib_Post( $this->post() );
+		$_post = $this->post();
+
+		if( $_post && 'post' !== $_post->post_type ){
+			WPLib::trigger_error( sprintf( "Current post is of '%s' post type and should not be requested via \$theme->post_item()", $_post->post_type) );
```

Comment:
@sarutole Should this trigger an error?  Are there no reasons why this might need to be called when it is invalid?